waf.go

```diff
@@ -95,3 +106,47 @@ func (api *API) ListWAFRules(zoneID, packageID string) ([]WAFRule, error) {
 	}
 	return rules, nil
 }
+
+// WAFRule returns a WAF rule from the given WAF package.
+//
+// API Reference: https://api.cloudflare.com/#waf-rules-rule-details
+func (api *API) WAFRule(zoneID, packageID, ruleID string) (WAFRule, error) {
+	var r WAFRuleResponse
+	uri := "/zones/" + zoneID + "/firewall/waf/packages/" + packageID + "/rules/" + ruleID
+	res, err := api.makeRequest("GET", uri, nil)
+	if err != nil {
+		return WAFRule{}, errors.Wrap(err, errMakeRequestError)
+	}
+
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return WAFRule{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return r.Result, nil
+}
+
+// UpdateWAFRule lets you update the mode of a WAF Rule.
+//
+// API Reference: https://api.cloudflare.com/#waf-rules-edit-rule
+func (api *API) UpdateWAFRule(zoneID, packageID, ruleID, mode string) (WAFRule, error) {
+	var r WAFRuleResponse
+	uri := "/zones/" + zoneID + "/firewall/waf/packages/" + packageID + "/rules/" + ruleID
+
+	params := struct {
```

Comment:
I'm not sure how I feel about this inline definition as it's not documented and inside of a function as opposed to exportable (like most of the other parts of this library). Was there a particular reason for not allowing this to be exported? 

There are quite a [few examples within this library](https://github.com/cloudflare/cloudflare-go/blob/89e1fce42324c0da504a3405f38e2a3910be7091/zone.go#L353-L358) where the PATCH requests rely on an exported struct and I'm thinking it might be best maintain that unless we've got a good reason to go against it.